### PR TITLE
SQLite RevEng: Ignore DEFAULT constraints for CLR defaults

### PR DESCRIPTION
**Pull request by bricelam**
Part of #9627

### CODE REVIEW

File: src/EFCore.Sqlite.Core/Scaffolding/Internal/SqliteDatabaseModelFactory.cs [hunk 1]
```diff
@@ -249,6 +255,28 @@ private IEnumerable<DatabaseColumn> GetColumns(DbConnection connection, string t
...
+                {
+                    var normalizedType = _typeMapper.FindMapping(dataType).StoreType;
+                    if (normalizedType == "INTEGER"
+                        || normalizedType == "REAL")
+                    {
+                        return null;
+                    }
+                }
+            }
+            else if (string.Equals(defaultValue, "null", StringComparison.OrdinalIgnoreCase))
```
Comment (bricelam) [inline src/EFCore.Sqlite.Core/Scaffolding/Internal/SqliteDatabaseModelFactory.cs]: Do this always, regardless. Using the `DEFAULT` results in an error anyway when `NOT NULL`.

Comment (bricelam) [reply src/EFCore.Sqlite.Core/Scaffolding/Internal/SqliteDatabaseModelFactory.cs]: (Discussed with @smitpatel)

File: src/EFCore.Sqlite.Core/Scaffolding/Internal/SqliteDatabaseModelFactory.cs [hunk 2]
```diff
@@ -249,6 +255,28 @@ private IEnumerable<DatabaseColumn> GetColumns(DbConnection connection, string t
             }
         }
 
+        private string IgnoreClrDefault(string dataType, bool notNull, string defaultValue)
```
Comment (smitpatel) [inline src/EFCore.Sqlite.Core/Scaffolding/Internal/SqliteDatabaseModelFactory.cs]: May be `FilterClrDefaultValue`?

File: test/EFCore.Sqlite.FunctionalTests/Scaffolding/SqliteDatabaseModelFactoryTest.cs [hunk 3]
```diff
@@ -30,11 +35,21 @@ private void Test(string createSql, IEnumerable<string> tables, IEnumerable<stri
 
             try
             {
-                var databaseModelFactory = new SqliteDatabaseModelFactory(
-                    new DiagnosticsLogger<DbLoggerCategory.Scaffolding>(
-                        new ListLoggerFactory(Log),
-                        new LoggingOptions(),
-                        new DiagnosticListener("Fake")));
+                // NOTE: You may need to update AddEntityFrameworkDesignTimeServices() too
```
Comment (smitpatel) [inline test/EFCore.Sqlite.FunctionalTests/Scaffolding/SqliteDatabaseModelFactoryTest.cs]: Pending note?

Comment (bricelam) [reply test/EFCore.Sqlite.FunctionalTests/Scaffolding/SqliteDatabaseModelFactoryTest.cs]: No, for future reference to others when they need to update it.

**Pull request merged**